pipelines/tekton/pipeplines_provider.go

```diff
@@ -97,16 +97,21 @@ func (pp *PipelinesProvider) Run(ctx context.Context, f fn.Function) error {
 		}
 	}
 
+	registry, err := docker.GetRegistry(f.Image)
+	if err != nil {
+		return err
+	}
+
 	_, err = k8s.GetSecret(ctx, getPipelineSecretName(f), pp.namespace)
 	if errors.IsNotFound(err) {
 		pp.progressListener.Stopping()
-		creds, err := pp.containerRegistryCredentialsCallback()
+		creds, err := pp.credentialsProvider(ctx, registry)
```

Comment:
Have you actually test this? I am not sure that registry == server` in this context. For `docker.io` registry a server `https://index.docker.io/v1/` was needed.

Reply:
I tested it with quay.io and it worked.

Reply:
that logic is also used for local push

Reply:
what happened when you used docker.io?

Reply:
maybe it's some quirk in `lifecycle` binary? Locally just `docker.io` works fine.

Reply:
in general `docker.io` is weird and treated as special case :disappointed:

Reply:
see https://github.com/google/go-containerregistry/blob/main/pkg/authn/keychain.go#L116-L122

Reply:
Yeah, it didn't work for `docker.io`.
Cool, let me try that.